lms/services/canvas_api.py

```diff
@@ -68,10 +68,13 @@ def list_files(self, course_id):
 
         :rtype: list(dict)
         """
-        return self._helper.validated_response(
-            self._helper.list_files_request(self._oauth2_token.access_token, course_id),
+        oauth2_token = self._oauth2_token
```

Comment:
If the refresh token used is always `self._oauth2_token` then that lookup could be pushed down into `send_with_refresh_and_retry`.

Reply:
Except that would cause two DB queries to happen instead of one. This could be optimised in a nicer way but for now this is how we avoid querying for the OAuth2Token twice